src/components/EditActivityPage/EditActivityPage.js

```diff
@@ -3,26 +3,26 @@ import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 
 import ActivityForm from '../ActivityForm/ActivityForm';
-import { editActivity, removeActivity } from '../../store/activities/actions';
+import { startEditActivity, startRemoveActivity } from '../../store/activities/actions';
 import { getEditedActivity } from '../../store/activities/selectors';
 import { ACTIVITY_PLAN_ROUTE } from '../../constants/routes';
 import activityPropTypeShape from '../../prop-types/activity';
 import historyPushPropTypeShape from '../../prop-types/history';
 
-const EditActivityPage = props => (
+const EditActivityPage = ({ activity, history, onStartEditActivity, onStartRemoveActivity }) => (
   <div>
     <ActivityForm
-      activity={props.activity}
-      onSubmit={activity => {
-        props.onEditActivity(props.activity.id, activity);
-        props.history.push(ACTIVITY_PLAN_ROUTE);
+      activity={activity}
+      onSubmit={activitySubmited => {
+        onStartEditActivity(activity.id, activitySubmited);
+        history.push(ACTIVITY_PLAN_ROUTE);
       }}
       data-test="form"
     />
     <button
       onClick={() => {
-        props.onRemoveActivity(props.activity.id);
-        props.history.push(ACTIVITY_PLAN_ROUTE);
+        onStartRemoveActivity(activity.id);
```

Comment:
I think that different scenarios should handled  (as well as in the previous onSubmit). 
At first... in this line, request is sent to the server. We are not sure, if this request was successful, so we don't know if we should redirect user to whatever is under `ACTIVITY_PLAN_ROUTE`. 

I suggest to:
- use async/await 
- use try... catch block
- this logic move . to the function using above points. 
- create a function before returning jsx
- wrap function with `useCallback` (read React docs and understand what it does). 

The same goes to the previous onSubmit.

Reply:
Ok, I forgot about one more task with this `redux-axios-middleware`. So, in next PR both of the issues should be handled :)

Reply:
Ok. I am changing my mind once again. 

`redux-axios-middleware` can stay for later PR, but this handling submission should be done during this one. Sorry, but I've seen already created actions ;)